pipelines/resources/tekton/task/func-deploy/0.1/func-deploy.yaml

```diff
@@ -24,7 +24,20 @@ spec:
       description: The workspace containing the function project
   steps:
   - name: func-deploy
-    image: "registry.redhat.io/openshift-serverless-1/kn-plugin-func-rhel8:latest"
+    image: "registry.access.redhat.com/ubi8/ubi"
```

Comment:
Shouldn't we use digest?

Reply:
and io registry?

Reply:
- The `registry.redhat.io/ubi8/ubi` requires [registry tokens](https://catalog.redhat.com/software/containers/ubi8/ubi/5c359854d70cc534b3a3784e?container-tabs=gti&gti-tabs=registry-tokens),
- The `registry.access.redhat.com/ubi8/ubi` can be used [without authentication](https://catalog.redhat.com/software/containers/ubi8/ubi/5c359854d70cc534b3a3784e?container-tabs=gti&gti-tabs=unauthenticated)
 
Most likely, we should stick with the `registry.access.redhat.com/ubi8/ubi` variant.

About the digest... That is the question, in fact!

Using the `registry.access.redhat.com/ubi8/ubi:latest` image will work, and has an advantage, that the image is always up-to-date. This automatically fixes all upcoming CVEs. On the other hand, if we decide to use SHA hash here, we'll be required to release Serverless just to fix this image, by bumping its SHA. Cumbersome, isn't?

Furthermore, in the disconnected clusters, this image needs to be [mirrored to bastion host](https://docs.openshift.com/container-platform/4.11/installing/disconnected_install/installing-mirroring-disconnected.html) to work. 

If we use the SHA hash, mirroring is simple. We have to put that image reference to the Operator's related images list. If we use the `:latest`, or a tag like `:8.6`, we could add the SHA hash of that image to the Operator's related images list, at the time of release. When mirroring is being done, the image will be mirrored by the hash, but will contain all assigned tags. This means the bastion registry will know all those assigned tags, not only the hash. This surely will work with the custom tags like `:8.6`. I have serious doubts whether it will work with the `:latest` tag. The `:latest` tag is ephemeral, assigned by registry.

**TL;DR:** Because of the above, I would vote to use `registry.access.redhat.com/ubi8/ubi:8.6`. 

For regular clusters, the image will be automatically updated, mitigating CVEs. For disconnected install, the image will be mirrored with a version that had `:8.6` tag at a time of our release. After a while, it may be outdated. It's not a big deal, though. Users could easily push the current `:8.6` image to the bastion host manually at a later date:

```shell
$ podman pull registry.access.redhat.com/ubi8/ubi:8.6
$ podman tag registry.access.redhat.com/ubi8/ubi:8.6 bastion.acme.localnet:5000/ubi8/ubi:8.6
$ podman push bastion.acme.localnet:5000/ubi8/ubi:8.6
```

WDYT? @jrangelramos @matejvasek @lance

Reply:
Although I like the idea of direct download from Serverless operator. I wonder if we should stick with more simple solution and try to use `kn-client` image for this release.

From @cardil summary, use of `registry.access.redhat.com/ubi8/ubi:8.6` sounds good to me. 

I wonder though if Tekton Pipelines doesn't provide a base "run my script" image that we could reuse here. Therefore we doesn't have to add any additional ubi image to our S-O `relatedImages` spec.